extensions/wikia/Forum/RelatedForumDiscussionController.class.php

```diff
@@ -77,14 +80,19 @@ public function checkData() {
 	 * @param int $threadId
 	 */
 	public static function purgeCache( $threadId ) {
-		$rm = new WallRelatedPages();
-		$ids = $rm->getMessagesRelatedArticleIds($threadId, 'order_index', DB_MASTER);
+		$relatedPages = new WallRelatedPages();
+		$ids = $relatedPages->getMessagesRelatedArticleIds( $threadId, 'order_index', DB_MASTER );
 
-		foreach($ids as $id) {
+		foreach( $ids as $id ) {
 			$key = wfMemcKey( __CLASS__, 'getData', $id );
-			WikiaDataAccess::cachePurge($key);
+			WikiaDataAccess::cachePurge( $key );
 			// VOLDEV-46: Update module by purging page, not via AJAX
-			WikiPage::newFromID( $id )->doPurge();
+			$wikiaPage = WikiPage::newFromID( $id );
+			if ( !empty( $wikiaPage ) ) {
+				$wikiaPage->doPurge();
+			} else {
+				self::logError( "doPurge() attempted on non-object", [ "articleID" => $id ] );
```

Comment:
"doPurge() attempted" might be misleading because it has not. Maybe something similar to "Found a null related wikipage on thread purge"?
I think threadId would also be very helpful in this log.